src/components/ui/button.tsx

```diff
@@ -60,4 +64,22 @@ const Button = React.forwardRef<HTMLButtonElement, ButtonProps>(
 );
 Button.displayName = "Button";
 
-export { Button, buttonVariants };
+const AuthorizedButton: React.FC<AuthorizedElementProps & ButtonProps> = ({
+  authorizeFor = () => true,
+  ...props
+}) => {
+  return (
+    <AuthorizedChild authorizeFor={authorizeFor}>
+      {({ isAuthorized }) => (
+        <Button
+          {...props}
+          disabled={props.disabled || !isAuthorized} // Disable if unauthorized or loading
```

Comment:
code is self explanatory.
```suggestion
          disabled={props.disabled || !isAuthorized}
```